Objects/longobject.c

```diff
@@ -4486,13 +4486,18 @@ divmod_shift(PyObject *shiftby, Py_ssize_t *wordshift, digit *remshift)
     return 0;
 }
 
+/* Inner function for both long_rshift and _PyLong_Rshift, shifting an
+   integer right by a nonnegative shift. */
```

Comment:
Could you add asserts that the shift is indeed non-negative.

Reply:
Done. I've added an `assert(wordshift>=0)` near the top, and removed a similar assert further down.